File: 1780-check-if-number-is-a-sum-of-powers-of-three/1780-check-if-number-is-a-sum-of-powers-of-three.java
class Solution {
    public boolean recur(int n , int x){
        if(n==0)return true;
        if(x>n)return false;
        return recur(n,x*3)||recur(n-x,x*3);
    }
    public boolean checkPowersOfThree(int n) {
        return recur(n,1);
    }
}

File: 1780-check-if-number-is-a-sum-of-powers-of-three/1780-check-if-number-is-a-sum-of-powers-of-three.cpp
class Solution {
public:
    bool f(int n,int power){
        if(n==0) return true;
        if(power>n) return false;
        return f(n,power*3) || f(n-power,power*3);
    }
    bool checkPowersOfThree(int n) {
        return f(n,1);
    }
};